src/System.Management.Automation/engine/PSVersionInfo.cs

```diff
@@ -53,20 +53,20 @@ public static class PSVersionInfo
         /// For each later release of PowerShell, this constant needs to
         /// be updated to reflect the right version.
         /// </remarks>
-        private static readonly Version s_psV1Version = new Version(1, 0);
-        private static readonly Version s_psV2Version = new Version(2, 0);
-        private static readonly Version s_psV3Version = new Version(3, 0);
-        private static readonly Version s_psV4Version = new Version(4, 0);
-        private static readonly Version s_psV5Version = new Version(5, 0);
-        private static readonly Version s_psV51Version = new Version(5, 1, NTVerpVars.PRODUCTBUILD, NTVerpVars.PRODUCTBUILD_QFE);
-        private static readonly SemanticVersion s_psV6Version = new SemanticVersion(6, 0, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV61Version = new SemanticVersion(6, 1, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV62Version = new SemanticVersion(6, 2, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV7Version = new SemanticVersion(7, 0, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV71Version = new SemanticVersion(7, 1, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psV72Version = new SemanticVersion(7, 2, 0, preReleaseLabel: null, buildLabel: null);
-        private static readonly SemanticVersion s_psSemVersion;
+        private static readonly Version s_psV1Version = new(1, 0);
+        private static readonly Version s_psV2Version = new(2, 0);
```

Comment:
Time to remove? :-)

Reply:
I do have changes in my local repo that removed all the end-of-life versions. I even think we should remove the `PSCompatibleVersions` key from `$PSVersionTable`. But the discussion should start with an issue.